File: frontend/types/aluno.ts
export interface Aluno {
  id: string;
  nome: string;
  email: string;
  turmaId: string | null;
  observacao: string | null;
}

/** Payload de criação — o backend exige turma no cadastro (`turma_id` não é opcional). */
export interface AlunoInput {
  nome: string;
  email: string;
  turmaId: string;
}

/** Payload de edição — aqui a turma pode ser desmarcada ("Sem turma"). */
export interface AlunoAtualizacaoInput {
  nome: string;
  email: string;
  turmaId: string | null;
  observacao: string | null;
}

File: backend/app/views/aluno.py
from app.models.aluno import Aluno
from app.schemas.aluno import AlunoOut


def to_aluno_out(aluno: Aluno) -> AlunoOut:
    return AlunoOut(id=aluno.id, nome=aluno.nome, email=aluno.email, turma_id=aluno.turma_id)
